src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -486,6 +520,91 @@ public function update( $type, $item ) {
 			}
 		}
 
+		if ( 'theme' === $type ) {
+			error_log( '    Theme ' . var_export( $item->theme, true ) . ' has been upgraded.' );
+		}
+
+		if ( 'plugin' === $type ) {
+			error_log( '    Plugin ' . var_export( $item->slug, true ) . ' has been upgraded.' );
+			if ( is_plugin_inactive( $upgrader_item ) ) {
+				error_log( '    ' . var_export( $upgrader_item, true ) . ' is inactive and will not be checked for fatal errors.' );
+			}
+
+			if ( $was_active && ! is_wp_error( $upgrade_result ) ) {
+
+				/*
+				 * The usual time limit is five minutes. However, as a loopback request
+				 * is about to be performed, increase the time limit to account for this.
+				 */
+				if ( function_exists( 'set_time_limit' ) ) {
+					set_time_limit( 10 * MINUTE_IN_SECONDS );
+				}
+
+				/*
+				 * Avoids a race condition when there are 2 sequential plugins that have fatal errors.
+				 * WP_Automatic_Updater::update() can be called for the second plugin before the first has
+				 * finished checking for fatal errors. This can cause the second plugin's fatal error checking to be skipped,
+				 * and may affect subsequent plugins too.
+				 */
```

Comment:
```suggestion
				/*
				 * Avoids a race condition when there are 2 sequential plugins that have fatal errors.
				 * It seems a slight delay is required for the loopback to use the updated plugin in the request.
				 * This can cause the second plugin's fatal error checking to be inaccurate,
				 * and may affect subsequent plugins too.
				 * Invalidating the updated plugin files using `wp_opcache_invalidate_directory()` has no effect, nor does a delay of less than 2 seconds.
				 */
```

Reply:
I'm not sure that documenting what doesn't work is useful here, but posting what's veen checked and verified is worth documenting on the ticket for posterity. We should document the reasoning for the sleep, and ensure that we've verified the details as much as we can

Reply:
By all means modify as necessary.

Reply:
Updated